### PR TITLE
Add hkl recipe

### PR DESCRIPTION
**Pull request by mrakitin**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mrakitin** [comment]
@prjemian, as we discussed earlier in person, I was about to create a conda-forge feedstock for the package. Finally, it's here in the staged recipes repo. Please accept the responsibility of maintaining it if you are still interested. Thanks!

**Comment by mrakitin** [comment]
I am not sure why the build is failing. When I run it locally, everything passes:
```
Packaging hkl
INFO:conda_build.build:Packaging hkl
INFO conda_build.build:build(1571): Packaging hkl
Packaging hkl-5.0.0.2173-py37hf484d3e_0
INFO:conda_build.build:Packaging hkl-5.0.0.2173-py37hf484d3e_0
INFO conda_build.build:bundle_conda(891): Packaging hkl-5.0.0.2173-py37hf484d3e_0
number of files: 10
Warning: rpath /opt/conda/conda-bld/hkl_1581378768894/_build_env/lib is outside prefix /opt/conda/conda-bld/hkl_1581378768894/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold (removing it)
WARNING (hkl,lib/libhkl.so.5.0.0): $RPATH/libopenblas.so.0 not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
WARNING (hkl,lib/libhkl.so.5.0.0): $RPATH/libgobject-2.0.so.0 not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
   INFO (hkl,lib/libhkl.so.5.0.0): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-hd420e75_5
WARNING (hkl,lib/libhkl.so.5.0.0): $RPATH/libglib-2.0.so.0 not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
WARNING (hkl,lib/libhkl.so.5.0.0): $RPATH/libgsl.so.25 not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
   INFO (hkl,lib/libhkl.so.5.0.0): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-hd420e75_5
WARNING (hkl): dso library package conda-forge::python-3.7.6-h357f687_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
Fixing permissions
Packaged license file/s.
Detected hard-coded path in text file lib/pkgconfig/hkl.pc
Importing conda-verify failed.  Please be sure to test your packages.  conda install conda-verify to make this message go away.
WARNING:conda_build.build:Importing conda-verify failed.  Please be sure to test your packages.  conda install conda-verify to make this message go away.
WARNING conda_build.build:bundle_conda(1042): Importing conda-verify failed.  Please be sure to test your packages.  conda install conda-verify to make this message go away.
TEST START: /opt/conda/conda-bld/linux-64/hkl-5.0.0.2173-py37hf484d3e_0.tar.bz2
Nothing to test for: /opt/conda/conda-bld/linux-64/hkl-5.0.0.2173-py37hf484d3e_0.tar.bz2
Renaming work directory,  /opt/conda/conda-bld/hkl_1581378768894/work  to  /opt/conda/conda-bld/hkl_1581378768894/work_moved_hkl-5.0.0.2173-py37hf484d3e_0_linux-64_main_build_loop
# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:

anaconda upload /opt/conda/conda-bld/linux-64/hkl-5.0.0.2173-py37hf484d3e_0.tar.bz2

# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes

anaconda_upload is not set.  Not uploading wheels: []
####################################################################################
Resource usage summary:

Total time: 0:04:54.6
CPU usage: sys=0:00:02.5, user=0:00:20.6
Maximum memory usage observed: 1.2G
Total disk usage observed (not including envs): 7.7M
```

Docker image version:
```bash
$ docker images condaforge/linux-anvil-comp7
REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
condaforge/linux-anvil-comp7   latest              459f2da0819d        16 hours ago        1.7GB
```

@conda-forge/staged-recipes, any ideas?

**Comment by isuruf** [comment]
Please see https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#requirements-section and fix the sections for the dependencies. Let me know if you need more help.

**Comment by mrakitin** [comment]
Thanks for the link, @isuruf! I rearranged some dependencies, and now there are less errors:
```py
Packaging hkl
INFO conda_build.build:build(1571): Packaging hkl
INFO conda_build.build:bundle_conda(891): Packaging hkl-5.0.0.2173-py36h618e01b_0
Packaging hkl-5.0.0.2173-py36h618e01b_0
Warning: rpath /home/conda/staged-recipes/build_artifacts/hkl_1581381928044/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/hkl_1581381928044/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho (removing it)
   INFO (hkl,lib/libhkl.so.5.0.0): Needed DSO lib/libgsl.so.23 found in conda-forge::gsl-2.5-h294904e_1
  ERROR (hkl,lib/libhkl.so.5.0.0): Needed DSO lib/libglib-2.0.so.0 found in ['glib']
  ERROR (hkl,lib/libhkl.so.5.0.0): .. but ['glib'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (hkl,lib/libhkl.so.5.0.0): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-hd420e75_5
  ERROR (hkl,lib/libhkl.so.5.0.0): Needed DSO lib/libgobject-2.0.so.0 found in ['glib']
  ERROR (hkl,lib/libhkl.so.5.0.0): .. but ['glib'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (hkl,lib/libhkl.so.5.0.0): Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libm.so.6 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-hd420e75_5
WARNING (hkl): run-exports library package conda-forge::pcre-8.43-he1b5a44_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (hkl): dso library package conda-forge::python-3.6.7-h357f687_1006 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
   INFO (hkl): plugin library package conda-forge::pygobject-3.30.5-py36h91a8f50_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (hkl): dso library package conda-forge::libstdcxx-ng-9.2.0-hdf63c60_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (hkl): run-exports library package conda-forge::libiconv-1.15-h516909a_1005 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
Traceback (most recent call last):
  File "/home/conda/.ci_support/build_all.py", line 135, in <module>
    build_all(args.recipes_dir, args.arch)
  File "/home/conda/.ci_support/build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File "/home/conda/.ci_support/build_all.py", line 125, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2344, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1643, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1002, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 867, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1150, in post_build
    check_overlinking(m, files)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1087, in check_overlinking
    m.config.enable_static)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1051, in check_overlinking_impl
    raise OverLinkingError(overlinking_errors)
conda_build.exceptions.OverLinkingError: overlinking check failed 
["  ERROR (hkl,lib/libhkl.so.5.0.0): .. but ['glib'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (hkl,lib/libhkl.so.5.0.0): .. but ['glib'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]
##[error]Bash exited with code '1'.
```
Added the `glib` requirement into the `host` section, I hope it helps.

**Comment by mrakitin** [comment]
What kind of tests can be done with this type of packages?

**Comment by mrakitin** [comment]
That helped, thanks!

As for the appveyor failure, it seems to be unrelated (EOL py27?):
```py
Checking to see if this PR build is outdated.
Traceback (most recent call last):
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 201, in <module>
    sys.exit(main())
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 192, in main
    exit_code = int(appveyor_check_latest_pr_build(repo, pr, bld) is False)
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 117, in appveyor_check_latest_pr_build
    data = request_json(url.format(repo=repo, total_builds=total_builds), headers=headers)
  File "C:\projects\staged-recipes\ff_ci_pr_build.py", line 46, in request_json
    with contextlib.closing(urlopen(request)) as response:
  File "C:\Python27\lib\urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Python27\lib\urllib2.py", line 429, in open
    response = self._open(req, data)
  File "C:\Python27\lib\urllib2.py", line 447, in _open
    '_open', req)
  File "C:\Python27\lib\urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "C:\Python27\lib\urllib2.py", line 1241, in https_open
    context=self._context)
  File "C:\Python27\lib\urllib2.py", line 1198, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)>
Command exited with code 1
```

**Comment by mrakitin** [comment]
This seems to be good to go. Just need a confirmation from @prjemian.

**Comment by prjemian** [comment]
I'll be a maintainer, too.

**Comment by mrakitin** [comment]
Thank you, @prjemian!

**Comment by prjemian** [comment]
Tests look very basic for packaging.  Functional tests would be much more complex.  Could be added later as enhancements.

Last two punch list items (above) can be checked off now.  Good to merge.

**Comment by mrakitin** [comment]
I agree about the tests. We'll need to extend them, maybe via the future `hklpy` package to be added here as well.

**Comment by mrakitin** [comment]
@isuruf, the problem is solved. Do you mind to review and merge if everything looks OK? Thanks.

File: recipes/hkl/build.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export PKG_CONFIG_LIBDIR="$PREFIX/lib/pkgconfig"
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$PKG_CONFIG_LIBDIR
+export ACLOCAL_PATH="$PREFIX/share/aclocal"
+aclocal --print-ac-dir
+
+autoreconf -ivf
+
+./configure --disable-gui --enable-introspection=yes --disable-hkl-doc --prefix=$PREFIX 
+# || { cat config.log ; exit 1 ; }
+make -j
```
Comment (scopatz) [inline recipes/hkl/build.sh]: This should be `make -j ${CPU_COUNT}`

File: recipes/hkl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - python
+    - gsl
+    - libiconv
+    - numpy
+    - pcre
+    - pygobject
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libhkl.a
```
Comment (scopatz) [inline recipes/hkl/meta.yaml]: Please do not package the static library

File: recipes/hkl/yum_requirements.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# gtk-doc provides "gtkdocize" needed during the building process
+gtk-doc
+# autoconf provides "autoreconf" needed during the building process, but it
+# also exists in the defaults channel of conda
+# autoconf
```
Comment (scopatz) [inline recipes/hkl/yum_requirements.txt]: Please remove the autoconf line

File: recipes/hkl/yum_requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# gtk-doc provides "gtkdocize" needed during the building process
+gtk-doc
```
Comment (scopatz) [inline recipes/hkl/yum_requirements.txt]: This seems like the sort of thing that should be a conda-forge package and listed in the build or host requirements

Comment (mrakitin) [reply recipes/hkl/yum_requirements.txt]: Can you please point which package would provide that? I didn't find any.

File: recipes/hkl/build.sh (hunk 1 above)
Comment (mrakitin) [reply recipes/hkl/build.sh]: Why is it bad? Too much stress on the CI?
```
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
```

File: recipes/hkl/yum_requirements.txt (hunk 3 above)
Comment (mrakitin) [reply recipes/hkl/yum_requirements.txt]: Done.

**Comment by mrakitin** [comment]
Here is the `./configure --help` output for our convenience.
<details>
<summary>output</summary>
<p>

```
`configure' configures hkl 5.0.0.0 to adapt to many kinds of systems.

Usage: ./configure [OPTION]... [VAR=VALUE]...

To assign environment variables (e.g., CC, CFLAGS...), specify them as
VAR=VALUE.  See below for descriptions of some of the useful variables.

Defaults for the options are specified in brackets.

Configuration:
  -h, --help              display this help and exit
      --help=short        display options specific to this package
      --help=recursive    display the short help of all the included packages
  -V, --version           display version information and exit
  -q, --quiet, --silent   do not print `checking ...' messages
      --cache-file=FILE   cache test results in FILE [disabled]
  -C, --config-cache      alias for `--cache-file=config.cache'
  -n, --no-create         do not create output files
      --srcdir=DIR        find the sources in DIR [configure dir or `..']

Installation directories:
  --prefix=PREFIX         install architecture-independent files in PREFIX
                          [/usr/local]
  --exec-prefix=EPREFIX   install architecture-dependent files in EPREFIX
                          [PREFIX]

By default, `make install' will install all the files in
`/usr/local/bin', `/usr/local/lib' etc.  You can specify
an installation prefix other than `/usr/local' using `--prefix',
for instance `--prefix=$HOME'.

For better control, use the options below.

Fine tuning of the installation directories:
  --bindir=DIR            user executables [EPREFIX/bin]
  --sbindir=DIR           system admin executables [EPREFIX/sbin]
  --libexecdir=DIR        program executables [EPREFIX/libexec]
  --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
  --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
  --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
  --libdir=DIR            object code libraries [EPREFIX/lib]
  --includedir=DIR        C header files [PREFIX/include]
  --oldincludedir=DIR     C header files for non-gcc [/usr/include]
  --datarootdir=DIR       read-only arch.-independent data root [PREFIX/share]
  --datadir=DIR           read-only architecture-independent data [DATAROOTDIR]
  --infodir=DIR           info documentation [DATAROOTDIR/info]
  --localedir=DIR         locale-dependent data [DATAROOTDIR/locale]
  --mandir=DIR            man documentation [DATAROOTDIR/man]
  --docdir=DIR            documentation root [DATAROOTDIR/doc/hkl]
  --htmldir=DIR           html documentation [DOCDIR]
  --dvidir=DIR            dvi documentation [DOCDIR]
  --pdfdir=DIR            pdf documentation [DOCDIR]
  --psdir=DIR             ps documentation [DOCDIR]

Program names:
  --program-prefix=PREFIX            prepend PREFIX to installed program names
  --program-suffix=SUFFIX            append SUFFIX to installed program names
  --program-transform-name=PROGRAM   run sed PROGRAM on installed program names

System types:
  --build=BUILD     configure for building on BUILD [guessed]
  --host=HOST       cross-compile to build programs to run on HOST [BUILD]

Optional Features:
  --disable-option-checking  ignore unrecognized --enable/--with options
  --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
  --enable-FEATURE[=ARG]  include FEATURE [ARG=yes]
  --enable-silent-rules   less verbose build output (undo: "make V=1")
  --disable-silent-rules  verbose build output (undo: "make V=0")
  --enable-dependency-tracking
                          do not reject slow dependency extractors
  --disable-dependency-tracking
                          speeds up one-time build
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-static[=PKGS]  build static libraries [default=yes]
  --enable-fast-install[=PKGS]
                          optimize for fast installation [default=yes]
  --disable-libtool-lock  avoid locking (might break parallel builds)
  --disable-gsltest       Do not try to compile and run a test GSL program
  --disable-glibtest      do not try to compile and run a test GLIB program
  --enable-logging        enable the glib logging
  --disable-hkl-doc       build documentation
  --enable-gtk-doc        use gtk-doc to build documentation [[default=no]]
  --enable-hkl3d          compile the hkl3d library
  --disable-gui           do not compile the gui interface
  --enable-introspection=[no/auto/yes]
                          Enable introspection for this build
  --enable-contrib        compile the contrib part

Optional Packages:
  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
  --with-pic[=PKGS]       try to use only PIC/non-PIC objects [default=use
                          both]
  --with-aix-soname=aix|svr4|both
                          shared library versioning (aka "SONAME") variant to
                          provide on AIX, [default=aix].
  --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
  --with-sysroot[=DIR]    Search for dependent libraries within DIR (or the
                          compiler's sysroot if not specified).
  --with-gsl-prefix=PFX   Prefix where GSL is installed (optional)
  --with-gsl-exec-prefix=PFX Exec prefix where GSL is installed (optional)
  --with-html-dir=PATH    path to installed docs

Some influential environment variables:
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  CXX         C++ compiler command
  CXXFLAGS    C++ compiler flags
  CPP         C preprocessor
  LT_SYS_LIBRARY_PATH
              User-defined run-time library search path.
  CXXCPP      C++ preprocessor
  PKG_CONFIG  path to pkg-config utility
  PKG_CONFIG_PATH
              directories to add to pkg-config's search path
  PKG_CONFIG_LIBDIR
              path overriding pkg-config's built-in search path
  GLIB_CFLAGS C compiler flags for GLIB, overriding pkg-config
  GLIB_LIBS   linker flags for GLIB, overriding pkg-config
  GLIB_GENMARSHAL
              value of glib_genmarshal for glib-2.0, overriding pkg-config
  GOBJECT_QUERY
              value of gobject_query for glib-2.0, overriding pkg-config
  GLIB_MKENUMS
              value of glib_mkenums for glib-2.0, overriding pkg-config
  GLIB_COMPILE_RESOURCES
              value of glib_compile_resources for gio-2.0, overriding
              pkg-config
  G3D_CFLAGS  C compiler flags for G3D, overriding pkg-config
  G3D_LIBS    linker flags for G3D, overriding pkg-config
  BULLET_CFLAGS
              C compiler flags for BULLET, overriding pkg-config
  BULLET_LIBS linker flags for BULLET, overriding pkg-config
  GTK_CFLAGS  C compiler flags for GTK, overriding pkg-config
  GTK_LIBS    linker flags for GTK, overriding pkg-config
  GLU_CFLAGS  C compiler flags for GLU, overriding pkg-config
  GLU_LIBS    linker flags for GLU, overriding pkg-config
  GTKGLEXT_CFLAGS
              C compiler flags for GTKGLEXT, overriding pkg-config
  GTKGLEXT_LIBS
              linker flags for GTKGLEXT, overriding pkg-config
  GOBJECT_CFLAGS
              C compiler flags for GOBJECT, overriding pkg-config
  GOBJECT_LIBS
              linker flags for GOBJECT, overriding pkg-config
  HDF5_CFLAGS C compiler flags for HDF5, overriding pkg-config
  HDF5_LIBS   linker flags for HDF5, overriding pkg-config

Use these variables to override the choices made by `configure' or to help
it to find libraries and programs with nonstandard names/locations.

Report bugs to <picca@synchrotron-soleil.fr>.
```

</p>
</details>

File: recipes/hkl/yum_requirements.txt (hunk 4 above)
Comment (scopatz) [reply recipes/hkl/yum_requirements.txt]: It seems like it is just called gtk-doc: https://www.gtk.org/gtk-doc/download.php

Comment (mrakitin) [reply recipes/hkl/yum_requirements.txt]: I saw that. Is there already a corresponding conda package for that? With this PR I didn't intend to build that package, but I can look into it.

Comment (scopatz) [reply recipes/hkl/yum_requirements.txt]: There doesn't seem to be a package for it yet: https://anaconda.org/search?q=gtk-doc

If you want to skip this step of the build, you can get away with not having this dep :wink:

File: recipes/hkl/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    - glib
+    - gobject-introspection
+    - gsl
+    - pygobject
+    - python
+  run:
+    - python
+    - gsl
+    - libiconv
+    - libopenblas
```
Comment (isuruf) [inline recipes/hkl/meta.yaml]: See docs on BLAS on how to correctly do this.

Comment (mrakitin) [reply recipes/hkl/meta.yaml]: What do you mean here? I don't think the package depends on `libopenblas`, I just wanted to get rid of the warnings in the packing step.

Comment (mrakitin) [reply recipes/hkl/meta.yaml]: @isuruf, I'd like to finish with this PR, but didn't quite get your points. Could you please clarify how it should be done? I was trying to get rid of the warnings in the build logs, which actually didn't break the builds. Any suggestions are much appreciated. Thank you!

**Comment by mrakitin** [comment]
Dear @conda-forge/staged-recipes maintainers, this PR seems to be passing well now. Can it me merged please as this package will very soon be needed for the deployment at NSLS-II? Thanks in advance!

File: recipes/hkl/meta.yaml (hunk 5 above)
Comment (isuruf) [reply recipes/hkl/meta.yaml]: See https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

File: recipes/hkl/build.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -e
+
+test -d m4 || mkdir m4
+# gtkdocize || exit 1
+
+export python=$PYTHON
+export PATH=$PREFIX/bin:$PATH
```
Comment (isuruf) [inline recipes/hkl/build.sh:9]: Why do you need this?

File: recipes/hkl/build.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+export PKG_CONFIG_LIBDIR="$PREFIX/lib/pkgconfig"
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$PKG_CONFIG_LIBDIR
```
Comment (isuruf) [inline recipes/hkl/build.sh:11]: Why do you need the two lines above?

File: recipes/hkl/COPYING [hunk 8]
```diff
@@ -0,0 +1,675 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (isuruf) [inline recipes/hkl/COPYING:1]: Why do you need this file here?

Comment (mrakitin) [reply recipes/hkl/COPYING:1]: To make sure it's packaged. Is it a bad practice?

File: recipes/hkl/build.sh (hunk 6 above)
Comment (mrakitin) [reply recipes/hkl/build.sh:9]: There was something bad going on with the build. I can remove it and try again.

File: recipes/hkl/COPYING (hunk 8 above)
Comment (isuruf) [reply recipes/hkl/COPYING:1]: Yes, it's a bad practice. See docs on how to include the license file from source.

File: recipes/hkl/build.sh (hunk 7 above)
Comment (mrakitin) [reply recipes/hkl/build.sh:11]: Good question. The first line came in historically from https://github.com/NSLS-II/lightsource2-recipes/blob/f5ba7dbac70a519f3a96720f4e017a7f6f021c93/recipes-tag/hkl/build.sh#L8. I think I had to add the second line during the course of the build to fix some issues (don't remember frankly speaking, was too long ago). I can run the build with the cleaned `build.sh` and see what's going on now.

File: recipes/hkl/COPYING (hunk 8 above)
Comment (mrakitin) [reply recipes/hkl/COPYING:1]: OK, I'll remove it from here. So, the assumption is that the license file should be present in the source distribution conda pulls during the building process and then this file is packaged, not the one which is provided, right?

Comment (mrakitin) [reply recipes/hkl/COPYING:1]: I was misled by many feedstocks having a license file in their `recipe` dirs. That's easier to maintain if it's not included here 😃

File: recipes/hkl/meta.yaml (hunk 5 above)
Comment (mrakitin) [reply recipes/hkl/meta.yaml]: That's an informative link, thanks! I saw you pushed a commit https://github.com/conda-forge/staged-recipes/pull/10819/commits/13952894632011dafad0be6ecfbaf53eb53844a7, so I assume this is resolved.

**Comment by mrakitin** [comment]
OK, first commented, and then removed unnecessary lines via a subsequent commit. Seems to build fine now! Thanks @isuruf for your suggestions! Anything else here or can it go in?

File: recipes/hkl/build.sh (hunk 1 above)
Comment (scopatz) [reply recipes/hkl/build.sh]: It is standard for our recipes

File: recipes/hkl/build.sh [hunk 9]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -e
+
+test -d m4 || mkdir m4
+# gtkdocize || exit 1
+
+export ACLOCAL_PATH="$PREFIX/share/aclocal"
+aclocal --print-ac-dir
```
Comment (isuruf) [inline recipes/hkl/build.sh:9]: Do you still need the above 5 lines?

File: recipes/hkl/build.sh [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+autoreconf -ivf
```
Comment (isuruf) [inline recipes/hkl/build.sh:11]:
```suggestion
chmod +x autogen.sh
./autogen.sh
```

File: recipes/hkl/build.sh (hunk 6 above)
Comment (mrakitin) [reply recipes/hkl/build.sh]: Removed now. Resolving.

File: recipes/hkl/meta.yaml (hunk 2 above)
Comment (mrakitin) [reply recipes/hkl/meta.yaml]: This was addressed.

File: recipes/hkl/build.sh (hunk 1 above)
Comment (mrakitin) [reply recipes/hkl/build.sh]: This is addressed. Thanks for the hint!

File: recipes/hkl/yum_requirements.txt (hunk 4 above)
Comment (mrakitin) [reply recipes/hkl/yum_requirements.txt]: The corresponding patch and the flags to `configure` exclude building docs for now.

File: recipes/hkl/build.sh (hunk 10 above)
Comment (mrakitin) [reply recipes/hkl/build.sh:11]: I wanted to use it originally, but https://github.com/picca/hkl/blob/e3215664bebb2c37020499c6636ebed4f65566c7/autogen.sh#L4 specifically requires `gtkdocize`, which is supplied via the `gtk-doc` package, which does not exist for conda yet. Also, `ACLOCAL_PATH` is not set in that script, so I'd rather keep the existing changes. Is it a compeling argument?

File: recipes/hkl/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - autoconf
+    - automake
+    - libtool
+    - m4
+    - pkg-config
```
Comment (scopatz) [inline recipes/hkl/meta.yaml:27]: please add make

File: recipes/hkl/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    - glib
+    - gobject-introspection
+    - gsl
+    - libcblas
+    - pygobject
+    - python
+  run:
+    - python
+    - gsl
+    - libiconv
```
Comment (scopatz) [inline recipes/hkl/meta.yaml:38]: libiconv probably needs to be in host too?

File: recipes/hkl/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    - numpy
+    - pcre
```
Comment (scopatz) [inline recipes/hkl/meta.yaml:40]: pcre also needs to be host

File: recipes/hkl/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+  license: GPL-3.0-or-later
+  license_file: COPYING
+  summary: Diffractometer computation library
+  doc_url: https://people.debian.org/~picca/hkl/hkl.html
+  dev_url: http://repo.or.cz/w/hkl.git
+
+extra:
+  recipe-maintainers:
+    - mrakitin
+    - prjemian
```
Comment (scopatz) [inline recipes/hkl/meta.yaml:59]: @prjemian - are you OK with being a maintainer?

**Comment by prjemian** [comment]
Yes

**Comment by mrakitin** [comment]
Hi @scopatz, your comments are addressed, and the build passes. Ready to go. Thanks!

**Comment by mrakitin** [comment]
Thanks @isuruf and @scopatz for the constructive feedback and merging this PR, and to @prjemian for agreeing to co-maintain the feedstock.

**Pull request merged**